projects/vdk-plugins/vdk-data-sources/src/vdk/plugin/data_sources/data_source.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from abc import abstractmethod
+from dataclasses import dataclass
+from dataclasses import field
+from datetime import datetime
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Union
+
+from vdk.plugin.data_sources.config import is_config_class
+from vdk.plugin.data_sources.state import IDataSourceState
+
+
+class IDataSourceConfiguration:
```

Comment:
what is the point in making an interface if it doesn't have any real methods in it?

Reply:
just add this check in the abstract class?

Reply:
What abstract class?